src/nvim/channel.c

```diff
@@ -372,10 +376,11 @@ uint64_t channel_connect(bool tcp, const char *address,
                          int timeout, const char **error)
 {
   Channel *channel;
+  bool loopback = false;
 
   if (!tcp && rpc) {
```

Comment:
TODO: writing an invalid message to a bytes (not RPC) "loopback" channel closes the stream, but not the channel, so `chansend()` still claims success but internally there's a SIGPIPE.

```vim
:let c = sockconnect('pipe', v:servername)
:echo chansend(c, "foo\n")
:echo chansend(c, "foo\n")
:edit $NVIM_LOG_FILE
```

Reply:
Hmm chansend should always check the stream, not the entire channel. Formally this is a pair of "normal" channels without any extra "loopback" logic, so I don't see what should go wrong here.